services-custom/s3-transfer-manager/src/main/java/software/amazon/awssdk/transfer/s3/internal/utils/ResumableFileDownloadSerializer.java

```diff
@@ -0,0 +1,269 @@
+        StructuredJsonGenerator generator = jsonGenerator();
+        JsonMarshallerRegistry marshallerRegistry = createMarshallerRegistry();
+        JsonMarshallerContext context = jsonMarshallerContext(generator, marshallerRegistry);
+
+        generator.writeStartObject();
+        SimpleTypeJsonMarshaller.LONG.marshall(resumableFileDownload.bytesTransferred(), context, "bytesTransferred", null);
+        SimpleTypeJsonMarshaller.INSTANT.marshall(resumableFileDownload.fileLastModified(), context, "fileLastModified", null);
+        if (resumableFileDownload.totalSizeInBytes().isPresent()) {
+            SimpleTypeJsonMarshaller.LONG.marshall(resumableFileDownload.totalSizeInBytes().get(),
+                                                   context,
+                                                   "totalSizeInBytes",
+                                                   null);
+        }
+        if (resumableFileDownload.s3ObjectLastModified().isPresent()) {
+            SimpleTypeJsonMarshaller.INSTANT.marshall(resumableFileDownload.s3ObjectLastModified().get(),
+                                                      context,
+                                                      "s3ObjectLastModified",
+                                                      null);
+        }
+        marshallDownloadFileRequest(resumableFileDownload.downloadFileRequest(), context);
+        generator.writeEndObject();
+
+        return generator.getBytes();
+    }
+
+    /**
+     * At this point we do not need to persist the TransferRequestOverrideConfiguration, because it only contains listeners and
+     * they are not used in the resume operation.
+     */
+    private static void marshallDownloadFileRequest(DownloadFileRequest fileRequest, JsonMarshallerContext context) {
+        context.jsonGenerator().writeFieldName("downloadFileRequest");
+        context.jsonGenerator().writeStartObject();
+        context.jsonGenerator().writeFieldName("destination");
+        context.jsonGenerator().writeValue(fileRequest.destination().toString());
+        marshallGetObjectRequest(fileRequest.getObjectRequest(), context);
+        context.jsonGenerator().writeEndObject();
+    }
+
+    private static void marshallGetObjectRequest(GetObjectRequest getObjectRequest, JsonMarshallerContext context) {
+        context.jsonGenerator().writeFieldName("getObjectRequest");
+        context.jsonGenerator().writeStartObject();
+        validateNoRequestOverrideConfiguration(getObjectRequest);
+        getObjectRequest.sdkFields().forEach(field -> marshallPojoField(field, getObjectRequest, context));
+        context.jsonGenerator().writeEndObject();
+    }
+
+    private static void validateNoRequestOverrideConfiguration(GetObjectRequest getObjectRequest) {
+        if (getObjectRequest.overrideConfiguration().isPresent()) {
+            log.warn(() -> "ResumableFileDownload GetObjectRequest contains an override configuration that will not be "
```

Comment:
Should it be `debug`? We can mention this behavior in the Javadocs

Reply:
Is that more consistent with our standard? What is our standard for these types of messages, would you say?